src/Microsoft.Management.Deployment/PackageManager.cpp

```diff
@@ -0,0 +1,316 @@
+#include "pch.h"
+#include "Public/AppInstallerCLICore.h"
+#include "Microsoft/PredefinedInstalledSourceFactory.h"
+#include "Commands/RootCommand.h"
+#include "ComContext.h"
+#include "ExecutionContext.h"
+#include "Workflows/WorkflowBase.h"
+#include <winget/UserSettings.h>
+#include "Commands/InstallCommand.h"
+#include <AppInstallerTelemetry.h>
+#include <AppInstallerErrors.h>
+#pragma warning( push )
+#pragma warning ( disable : 4467 6388)
+// 6388 Allow CreateInstance.
+#include <wil\cppwinrt_wrl.h>
+// 4467 Allow use of uuid attribute for com object creation.
+#include "PackageManager.h"
+#pragma warning( pop )
+#include "PackageManager.g.cpp"
+#include "InstallResult.h"
+#include "PackageCatalogInfo.h"
+#include "PackageCatalogReference.h"
+#include "PackageVersionInfo.h"
+#include "PackageVersionId.h"
+#include "Converters.h"
+#include "Helpers.h"
+
+using namespace std::literals::chrono_literals;
+
+const GUID PackageManagerCLSID1 = { 0xC53A4F16, 0x787E, 0x42A4, { 0xB3, 0x04, 0x29, 0xEF, 0xFB, 0x4B, 0xF5, 0x97 } };  //C53A4F16-787E-42A4-B304-29EFFB4BF597
+const GUID PackageManagerCLSID2 = { 0xE65C7D5A, 0x95AF, 0x4A98, { 0xBE, 0x5F, 0xA7, 0x93, 0x02, 0x9C, 0xEB, 0x56 } };  //E65C7D5A-95AF-4A98-BE5F-A793029CEB56
+
+namespace winrt::Microsoft::Management::Deployment::implementation
+{
+    winrt::Windows::Foundation::Collections::IVectorView<winrt::Microsoft::Management::Deployment::PackageCatalogReference> PackageManager::GetPackageCatalogs()
+    {
+        Windows::Foundation::Collections::IVector<Microsoft::Management::Deployment::PackageCatalogReference> catalogs{ winrt::single_threaded_vector<Microsoft::Management::Deployment::PackageCatalogReference>() };
+        std::vector<::AppInstaller::Repository::SourceDetails> sources = ::AppInstaller::Repository::GetSources();
+        for (uint32_t i = 0; i < sources.size(); i++)
+        {
+            auto packageCatalogInfo = winrt::make_self<wil::details::module_count_wrapper<winrt::Microsoft::Management::Deployment::implementation::PackageCatalogInfo>>();
+            packageCatalogInfo->Initialize(sources.at(i));
+            auto packageCatalogRef = winrt::make_self<wil::details::module_count_wrapper<winrt::Microsoft::Management::Deployment::implementation::PackageCatalogReference>>();
+            packageCatalogRef->Initialize(*packageCatalogInfo);
+            catalogs.Append(*packageCatalogRef);
+        }
+        return catalogs.GetView();
+    }
+    winrt::Microsoft::Management::Deployment::PackageCatalogReference PackageManager::GetPredefinedPackageCatalog(winrt::Microsoft::Management::Deployment::PredefinedPackageCatalog const& predefinedPackageCatalog)
+    {
+        switch (predefinedPackageCatalog)
+        {
+        case winrt::Microsoft::Management::Deployment::PredefinedPackageCatalog::OpenWindowsCatalog:
+            // TODO: Mapping of enum to sources should link directly to definition, not copy string.
+            return GetPackageCatalogByName(L"winget");
```

Comment:
Should be implemented with a new `OpenWellKnownSource` function (or similar name).  Could also use the existing `OpenPredefinedSource` and add new enum values, as well as changing the internals to not assume that the result is of the type Predefined.